sdks/java/core/src/test/java/org/apache/beam/sdk/io/LocalFileSystemTest.java

```diff
@@ -242,6 +246,52 @@ public void testMatchMultipleWithFileExtension() throws Exception {
   }
 
   @Test
+  public void testMatchInDirectory() throws Exception {
+    List<String> expected = ImmutableList.of(temporaryFolder.newFile("a").toString());
+    temporaryFolder.newFile("aa");
+    temporaryFolder.newFile("ab");
+
+    String expectedFile = expected.get(0);
+    int slashIndex = expectedFile.lastIndexOf('/');
+    if (SystemUtils.IS_OS_WINDOWS) {
+        slashIndex = expectedFile.lastIndexOf('\\');
+    }
+    String directory = expectedFile.substring(0, slashIndex);
+    String relative = expectedFile.substring(slashIndex + 1);
+    System.setProperty("user.dir", directory);
```

Comment:
Use RestoreSystemProperties or you'll modify the execution environment for other tests.